src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from "react";
+import { RadioGroup as ChakraRadioGroup, Stack } from "@chakra-ui/react";
+
+import HelperErrorText from "../HelperErrorText/HelperErrorText";
+import generateUUID from "../../helpers/generateUUID";
+import { spacing } from "../../theme/foundations/spacing";
+
+export enum RadioGroupLayouts {
+  Vertical = "vertical",
+  Horizontal = "horizontal",
+}
+
+type DirectionMap = { [key: string]: "column" | "row" };
+
+export interface RadioGroupProps {
+  /** The radio group label. */
+  labelText: string;
+  /** className you can add in addition to 'input' */
+  className?: string;
+  /** Adds the 'disabled' prop to the input when true */
+  disabled?: boolean;
+  /** Helper for modifiers array; adds 'errored' styling */
+  errored?: boolean;
+  required?: boolean;
+  optReqFlag?: boolean;
+  /** Optional string to populate the HelperErrorText for error state */
+  errorText?: string;
+  /** Optional string to populate the HelperErrorText for standard state */
+  helperText?: string;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  /** The name prop indicates into which group of radios this radio belongs.  If none is specified, 'default' will be used */
+  name?: string;
+  /** The action to perform on the `<input>`'s onChange function  */
+  onChange?: (value: string) => string;
+  /** Offers the ability to show the radio's label onscreen or hide it. Refer to the `labelText` property for more information. */
+  showLabel: boolean;
+  /** Populates the value of the input */
+  defaultValue?: string;
+  layout?: RadioGroupLayouts;
+}
+
+export const onChangeDefault = () => {
+  return;
+};
+
+const directionMap: DirectionMap = {
+  [RadioGroupLayouts.Vertical]: "column",
+  [RadioGroupLayouts.Horizontal]: "row",
+};
+
+function RadioGroup(props: React.PropsWithChildren<RadioGroupProps>) {
+  const {
+    className,
+    id = generateUUID(),
+    children,
+    name,
+    defaultValue,
+    onChange = onChangeDefault,
+    layout = RadioGroupLayouts.Vertical,
+    labelText,
+    helperText,
+    errorText,
+    showLabel,
+    errored = false,
+    required,
+    disabled,
+  } = props;
+  const baseName = "RadioGroup";
+  const footnote = errored ? errorText : helperText;
+  const direction = directionMap[layout];
+  const newChildren = [];
+
+  React.Children.map(children, (child: React.ReactElement, i) => {
```

Comment:
I mentioned in the spec doc that we should enforce consumers to use the DS `Radio` component within `RadioGroup` because we can easily control all the styling. Here, if the `RadioGroup` is errored or disabled, it's easy to pass the same "disabled" and "errored" props to the `Radio` component. If it was a native HTML element, "errored" will not work and we won't get the desired style.

Reply:
I do agree with you and would love to state that the DS `Radio` component is required, but I worry about pushback from comsuming app devs.

Reply:
I don't think it's too much to ask to use a DS `Radio` component. If someone is using Chakra, they would use all relevant components including Chakra's `RadioGroup` and `Radio`. If that's their pattern, I think we can follow it and say consumers _must_ use the DS `Radio` component within a `RadioGroup`.

Reply:
OK, I am on board with this.